### PR TITLE
feat(compiler)!: Make List a language-supplied type

### LINKED ISSUE #552: Runtime: Use real list implementation in toString
Let's remove this once we move List into a type that is built into the compiler.

_Originally posted by @phated in https://github.com/grain-lang/grain/pull/540#discussion_r589953117_

### LINKED ISSUE #1593: Move List type into the language
The List type should be a part of the language instead of Pervasives.

### PR DESCRIPTION
**Pull request by ospencer**
Closes #552 
Closes #1593

### CODE REVIEW

**Review by phated — CHANGES_REQUESTED** [review]
2 questions and I think we have a conflict on the generated docs.

File: stdlib/runtime/string.gr [hunk 1]
```diff
@@ -400,6 +396,8 @@ let rec heapValueToString = (ptr, extraIndents, toplevel) => {
           WasmI32.toGrain(builtinVariantName),
           extraIndents
         )
+      } else if (isListVariant(ptr)) {
+        listToString(ptr, extraIndents)
```
[the remark this answers is not in the document]
Comment (ospencer) [reply stdlib/runtime/string.gr:400]: `getBuiltinVariantName` tells you the name of the variant, so in the case of lists it's cons/empty, but variant name isn't a factor in printing lists.

File: stdlib/runtime/string.gr [hunk 2]
```diff
@@ -582,7 +575,7 @@ listToString = (ptr, extraIndents) => {
 
   while (true) {
     let variantId = WasmI32.load(cur, 12n) >> 1n // tagged number
-    if (variantId == 0n) {
+    if (variantId == 1n) {
```
[the remark this answers is not in the document]
Comment (ospencer) [reply stdlib/runtime/string.gr:578]: I changed the order of the variants for List, so the IDs swapped. There is no technical reason for this change; it was for my own self-satisfaction and personal philosophy on which of the two variants should come first.

**Review by phated — APPROVED** [review]
Excellent! Thanks 🎉

**Pull request merged**